src/components/models/Pryv.js

```diff
@@ -35,26 +37,50 @@ export type ServiceInfos = {
 }
 
 class Pryv {
-  core: (string) => string;
+  core: (string, ?string) => string;
+  serviceInfoUrl: string;
   register: string;
-
-  constructor (domain: string) {
-    this.core = username => `https://${username}.${domain}`;
-    this.register = `https://reg.${domain}`;
+  apiUrl: string;
+
+  constructor (serviceInfoUrl: string) {
+    this.serviceInfoUrl = serviceInfoUrl;
+    this.getServiceInfo()
+      .then(res => {
+        this.apiUrl = res.api;
+        this.register = res.register;
+      })
+      .catch(err => {
+        console.error('Problem while fetching service info : ', err);
+      });
+
+    this.core = function (username: string, path: ?string) {
+      if (this.apiUrl == null) {
+        console.error('apiUrl is not set yet');
+        return '';
+      }
+      path = path || '';
+      return url.resolve(this.apiUrl.replace('{username}', username), path);
+    };
   }
 
   // ---------- AUTH calls ----------
 
   // GET/reg: polling with according poll key
   async poll (pollKey: string): Promise<AuthState> {
-    const res = await axios.get(`${this.register}/access/${pollKey}`);
+    if (this.register == null) {
+      return Promise.reject(new Error('register url is not set yet, wait for service info to be loaded'));
+    }
```

Comment:
Est-ce qu'on fait vraiment le test à chaque fois ? Chacune de ces méthodes qui utilise this.register est déclanchée par un événement de l'utilisateur, donc la page aura eu le temps de fetch service info (dans le constructeur)